lib/cassy/cas.rb

```diff
@@ -158,14 +162,20 @@ def base_service_url(full_service_url)
     def detect_ticketing_service(service)
       # try to find the service in the valid_services list
       # if loosely_matched_services is true, try to match the base url of the service to one in the valid_services list
+      # if match_on_top_level_domain is true, ensure the base url of the service belongs to an allowed top level domain
       # if still no luck, check if there is a default_redirect_url that we can use
       @service ||= service
+      is_valid_service_domain = valid_domains.detect{|tld| validate_top_level_domain(tld, @service)}
```

Comment:
rename to `valid_service_domain`, maybe? I thought this was a boolean due to the name.